File: replace-all-digits-with-characters.ts
function replaceDigits(s: string): string {
  const alphabet = "abcdefghijklmnopqrstuvwxyz";
  let result = "";

  for (let i = 1; i < s.length; i += 2) {
    result += s[i - 1] + alphabet[+s[i] + alphabet.indexOf(s[i - 1])];
  }

  if (result.length < s.length) result += s[s.length - 1];

  return result;
}

File: replace-all-digits-with-characters.py
class Solution:
    def replaceDigits(self, s: str) -> str:
        alphabet = 'abcdefghijklmnopqrstuvwxyz'
        result = ''

        for i, c in enumerate(s):
            if i % 2 == 0:
                result += s[i]
            else:
                result += alphabet[int(s[i]) + alphabet.index(s[i - 1])]

        return result
